rustler_codegen/src/nif.rs

```diff
@@ -0,0 +1,205 @@
+use proc_macro2::{Span, TokenStream};
+use quote::{quote, ToTokens};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+
+pub fn transcoder_decorator(args: syn::AttributeArgs, fun: syn::ItemFn) -> TokenStream {
+    let sig = &fun.sig;
+    let name = &sig.ident;
+    let inputs = &sig.inputs;
+    let flags = schedule_flag(args.to_owned());
+    let function = fun.to_owned().into_token_stream();
+    let arity = arity(inputs.clone());
```

Comment:
`arity()` could also take `&Punctuated<_,_>`. Then, cloning is not needed. This is also possible with `extract_inputs` and `create_function_params`.

Reply:
I had all this at first, but then clippy was complaining:

```
writing `&Vec<_>` instead of `&[_]` involves one more reference and cannot be used with non-Vec-based slices.
```

Reply:
Interesting, clippy does not complain here when I try that (clippy 0.0.212 (3aea860 2019-09-03)). Feel free to resolve here and the related comments!